### PR TITLE
Upgrade GitHub Actions

### PR DESCRIPTION
**Pull request by cclauss**
Trying to upgrade to a current version of mypy
* https://github.com/actions/checkout/releases
* https://github.com/actions/setup-python/releases

### CODE REVIEW

File: .github/workflows/tox.yml [hunk 1]
```diff
@@ -1,16 +1,24 @@
...
+      - master
+
 jobs:
   tox:
     strategy:
       fail-fast: false
       matrix:
         os: [ubuntu-latest]
-        python: ['3.6', '3.7', '3.8', '3.9', '3.10']
+        python: ['3.6', '3.7', '3.8', '3.9', '3.10', '3.11']
```
Comment (cmurtaugh) [inline .github/workflows/tox.yml:17]: It looks like GitHub doesn't have a python 3.6 runner available for ubuntu-latest which is currently 22.04; I think we'd need to either drop 3.6 or pin ubuntu to 20.04.

Comment (cclauss) [reply .github/workflows/tox.yml:17]: Correct.  Which way should we go?
* https://devguide.python.org/versions

Comment (cmurtaugh) [reply .github/workflows/tox.yml:17]: My vote would be to retain support for 3.6 and pin ubuntu to 20.04 for now. That seems to me like it'd have the least negative impact right now for consumers of this library.

**Comment by cclauss** [comment]
GitHub Actions on current Ubuntu does not support Python 3.6 because the Python Core Team does not support Py36.
* https://devguide.python.org/versions

I am not a fan of encouraging folks to build solutions on unsupported platforms but here goes...

**Comment by cmurtaugh** [comment]
@cclauss that's a good point! I don't feel strongly about retaining support for 3.6 here; was really just trying not to rock the boat. :-)

**Comment by cclauss** [comment]
@cmurtaugh I stole you `pylint<2.15.6` to get things to pass.  I am not a fan of pylint.

**Pull request closed without merging**